client/src/lib/attempt-result.js

```diff
@@ -483,3 +489,15 @@ export function meetsCutoff(attemptResults, cutoff) {
     .slice(0, numberOfAttempts)
     .some((attempt) => attempt > 0 && attempt < attemptResult);
 }
+
+/**
+ * Checks if the given attempt results contain a DNS followed by a valid result.
+ */
+export function checkForDnsFollowedByValidResult(attemptResults) {
+  const dnsIndex = attemptResults.findIndex((attempt) => attempt === DNS_VALUE);
+  if (dnsIndex === -1) return false;
+  const validResultIndex = attemptResults.findIndex(
+    (attempt, index) => (attempt > 0 || attempt === -1) && index > dnsIndex
+  );
+  return validResultIndex !== -1;
```

Comment:
```suggestion
  return attemptResults.some(
    (attempt, index) => index > dnsIndex && (attempt !== SKIPPED_VALUE && attempt !== DNS_VALUE)
  );
```

Both checks work, but I think this is a bit more clear about the intent.